Team417_PowerPlay/src/main/java/org/firstinspires/ftc/team417_PowerPlay/BaseTeleOp.java

```diff
@@ -1,44 +1,149 @@
 package org.firstinspires.ftc.team417_PowerPlay;
 
+import com.qualcomm.robotcore.util.ElapsedTime;
+import com.qualcomm.robotcore.util.Range;
+
 abstract public class BaseTeleOp extends BaseOpMode {
-    boolean grabberOpen = false;
+    boolean grabberClosed = false;
     double armPower = 0.0;
+    int armLevel = 0;
+    int armEncoderPosition = 0;
+
+    boolean armManual = false;
+
+    ElapsedTime time;
+    private static double BUMPER_TIMEOUT = 0.3; // in seconds
+
+    private static double LOWER_ARM_MANUAL_POWER = -0.2;
+    private static double LOWER_ARM_AUTO_POWER = -0.4;
+    private static double RAISE_ARM_MANUAL_POWER = 1.0 / 100.0;
+    private static double RAISE_ARM_AUTO_POWER = 1.0 / 800.0;
 
-    private static double DRIVING_SPEED = 0.5;
+    private static double DRIVING_SPEED = 0.7;
+
+    public void initializeTeleOp() {
+        initializeHardware();
+        time = new ElapsedTime();
+        time.reset();
+
+        grabberServo.setPosition(GRABBER_OPEN);
+
+    }
```

Comment:
```suggestion
        grabberServo.setPosition(GRABBER_OPEN);
    }
```
https://google.github.io/styleguide/javaguide.html#s4.1-braces